Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -3465,9 +3492,14 @@ private unsafe void OnUpdate(InputUpdateType updateType, ref InputEventBuffer ev
 
                     // Update metrics.
                     if (currentEventTimeInternal <= currentTime)
-                        totalEventLag += currentTime - currentEventTimeInternal;
-                    ++m_Metrics.totalEventCount;
-                    m_Metrics.totalEventBytes += (int)currentEventReadPtr->sizeInBytes;
+                    {
+                        var lag = currentTime - currentEventTimeInternal;
+                        totalEventLag += lag;
+                        if (lag > maxEventLag)
+                            maxEventLag = lag;
+                    }
+                    ++totalEventCount;
+                    totalEventSizeBytes += (int)currentEventReadPtr->sizeInBytes;
```

Comment:
This a bit more complex so I would extract this to a method, as `OnUpdate` is already big enoug.